### PR TITLE
CCE Node: made iptype, bandwidth_charge_mode, sharetype settable

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
## Summary of the Pull Request
[CCE] iptype, bandwidth_charge_mode, sharetype node properties are not settable

## PR Checklist

* [x] Refers to: #665
* [x] Tests added/passed.
* [ ] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccCCENodesV3_basic
--- PASS: TestAccCCENodesV3_basic (1914.87s)
=== RUN   TestAccCCENodesV3_ip
--- PASS: TestAccCCENodesV3_ip (1885.70s)
=== RUN   TestAccCCENodesV3_ipWithExtendedParameters
--- PASS: TestAccCCENodesV3_ipWithExtendedParameters (1866.73s)
=== RUN   TestAccCCENodesV3_ipNulls
--- PASS: TestAccCCENodesV3_ipNulls (1939.93s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/diff_suppress_funcs.go [hunk 1]
```diff
@@ -42,8 +42,8 @@ func suppressMinDisk(k, old, new string, d *schema.ResourceData) bool {
 	return new == "0" || old == new
 }
 
-// Suppress changes if we don't specify an external gateway, but one is specified for us
-func suppressExternalGateway(k, old, new string, d *schema.ResourceData) bool {
+// Suppress changes if we don't specify it, but one is specified for us
+func suppressEmpty(k, old, new string, d *schema.ResourceData) bool {
```
Comment (lego963) [inline opentelekomcloud/diff_suppress_funcs.go]: what about to rename`suppressEmptyField` ?

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 2]
```diff
@@ -394,9 +397,15 @@ func resourceCCENodeV3Create(d *schema.ResourceData, meta interface{}) error {
 	eipCount := d.Get("eip_count").(int)
 	if bandwidthSize > 0 && eipCount == 0 {
 		eipCount = 1
-		_ = d.Set("bandwidth_charge_mode", "traffic")
-		_ = d.Set("sharetype", "PER")
-		_ = d.Set("iptype", "5_bgp")
+		if d.Get("bandwidth_charge_mode").(string) == "" {
+			_ = d.Set("bandwidth_charge_mode", "traffic")
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go:401]: Can the same be done with `DefaultFunc`?

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go [hunk 3]
```diff
@@ -267,3 +288,37 @@ resource "opentelekomcloud_cce_node_v3" "node_1" {
...
+  flavor_id         = "s2.xlarge.2"
+  availability_zone = "%s"
+  key_pair          = "%s"
+  root_volume {
+    size       = 40
+    volumetype = "SATA"
+  }
+
+  bandwidth_size = 100
+  sharetype      = "PER"
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go:334]: Have you tested with `null`s?

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go (hunk 2 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go:401]: DefaultFunc takes values from env vars, i don't think that we need that here

Comment (outcatcher) [reply opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go:401]: `DefaultFunc` can be any function you want:
`type SchemaDefaultFunc func() (interface{}, error)`
Though, it is impossible to access other resource values from it

Comment (anton-sidelnikov) [reply opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go:401]: Yes, it's that what i mean

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go (hunk 3 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go:334]: === RUN   TestAccCCENodesV3_ipNulls
--- PASS: TestAccCCENodesV3_ipNulls (1939.93s)
PASS

File: opentelekomcloud/diff_suppress_funcs.go (hunk 1 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/diff_suppress_funcs.go]: not using this suppress amymore

**Review by lego963 — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**